### PR TITLE
Added custom markdown processing

### PR DESCRIPTION
**Pull request by DevNico**
This should fix the relative img resolution in #4

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 unpacked-extension
 tako-github-file-tree.zip
-node_modules
\ No newline at end of file
+node_modules
```
Comment (brumm) [inline .gitignore]: Thanks! FYI: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

**Comment by brumm** [comment]
Thanks for your contribution! I might not get around to a proper review until after my day job, might take a couple of hours, fyi :)

File: .gitignore (hunk 1 above)
Comment (DevNico) [reply .gitignore]: That's interesting but I don't think it's something you should depend on in open source projects.

File: Preview.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
+import React from 'react'
```
Comment (brumm) [inline Preview.js:1]: This file seem duplicated and not in use

File: src/Preview.js [hunk 3]
```diff
@@ -36,15 +39,57 @@ const MarkdownPreview = ({ path }) => {
...
-  if (status === 'loading') {
-    return <LoadingComponent />
+  const staticUrlRegex = new RegExp(/^https?:/)
+
+  const transformImageUri = uri => {
+    if (!staticUrlRegex.test(uri)) {
+      return `https://github.com/${user}/${repo}/raw/${branch}/${uri}`
+    } else {
+      return uri.replace('/blob/', '/raw/').replace('http://', 'https://')
+    }
```
Comment (brumm) [inline src/Preview.js:50]: I love being able to walk to markdown AST and correcting the URLs!

What I'm concerned about though, is forgoing the github markdown redering endpoint and doing all of the complicated work ourselves. Github-flavored-markdown has become pretty complex, and we'd need to handle _all_ of the things it does if we decide to render it ourselves (code highlighting, commit-, branch-, user-, pr-, issue references formatting and linking etc.).

I'm unwilling to take on all that complexity just to fix some links. But! I figure we could have the cake and eat it too :)

I propose that we process and fix the links _before_ we send it to the GH markdown rendering service. That takes care of 100% of the complex stuff that GH is doing with GFM and we get to fix our srcs and hrefs too. What do you think?

**Review by brumm — CHANGES_REQUESTED** [review]

File: src/Preview.js (hunk 3 above)
Comment (DevNico) [reply src/Preview.js:50]: I totally agree with you, I didn't think about all the other magical unknown stuff GFM has and does. I'll update this PR in the next few days when I find the time to do so.

File: Preview.js (hunk 2 above)
Comment (DevNico) [reply Preview.js:1]: Will fix.

**Pull request closed without merging**